### PR TITLE
WIP: PoC: use pre-built tf provider binaries

### PR DESCRIPTION
**Pull request by r4f4**

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -37,7 +37,10 @@ fi
 export CGO_ENABLED=0
 MODE="${MODE:-release}"
 # build terraform binaries before setting environment variables since it messes up make
-make -C terraform all
+BUILD_TF_PROVIDERS="${BUILD_TF_PROVIDERS:-1}"
+if [ "${BUILD_TF_PROVIDERS}" -ne 0 ]; then
+	make -C terraform all
```
[the remark this answers is not in the document]
Comment (r4f4) [reply hack/build.sh:42]: @patrickdillon if the Installer repo is freshly cloned every time, then the timestamp of the files will be more recent than the ones in the terraform container and `make` will rebuild them. If `ci-operator` just fetches updates on an existing clone, then we can drop this change.

[the remark this answers is not in the document]
Comment (r4f4) [reply hack/build.sh:42]: @patrickdillon I haven't tried including the `go.sum`, let me experiment with the implications of that.

[the remark this answers is not in the document]
Comment (patrickdillon) [reply hack/build.sh:42]: my bad, it's go.mod

**Pull request closed without merging**